services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3339,6 +3338,30 @@ private OfflinePushStatusInfo getOffLineJobStatus(
               Optional.of(currentReturnStatusDetails.orElse("") + "Parent Kafka topic truncated");
         }
       }
+
+      // Update version status on parent
+      // COMPLETED -> ONLINE
+      // ERROR -> ERROR
+      // ERROR in some region but COMPLETED in others -> PARTIALLY_ONLINE
+      if (!incrementalPushVersion.isPresent()) {
+        final VersionStatus finalVersionStatusOnParent;
+        if (currentReturnStatus.equals(ExecutionStatus.COMPLETED)) {
+          finalVersionStatusOnParent = VersionStatus.ONLINE;
+        } else if (currentReturnStatus.equals(ExecutionStatus.ERROR) && successCount > 0) {
+          finalVersionStatusOnParent = VersionStatus.PARTIALLY_ONLINE;
+        } else {
+          finalVersionStatusOnParent = VersionStatus.ERROR;
+        }
+        final String storeName = Version.parseStoreFromKafkaTopicName(kafkaTopic);
```

Comment:
Can we verify this all comes up right in a test which spins up a parent controller?

Reply:
Good point! Will do.